test/helpers/customError.js

```diff
@@ -1,19 +1,37 @@
 const { config } = require('hardhat');
+const { expect } = require('chai');
 
 const optimizationsEnabled = config.solidity.compilers.some(c => c.settings.optimizer.enabled);
 
 /** Revert handler that supports custom errors. */
-async function expectRevertCustomError(promise, reason) {
+async function expectRevertCustomError(promise, expectedErrorName, args) {
   try {
     await promise;
     expect.fail("Expected promise to throw but it didn't");
   } catch (revert) {
-    if (reason) {
+    if (!Array.isArray(args)) {
+      expect.fail('Expected 3rd array parameter for error arguments');
+    }
+    if (expectedErrorName) {
       if (optimizationsEnabled) {
         // Optimizations currently mess with Hardhat's decoding of custom errors
-        expect(revert.message).to.include.oneOf([reason, 'unrecognized return data or custom error']);
+        expect(revert.message).to.include.oneOf([expectedErrorName, 'unrecognized return data or custom error']);
```

Comment:
This is saying that with optimizations enabled the custom errors are not checked, right?

That's kind of unfortunate because our `tests` check runs with optimizations, because at the same time it produces the gas report. The `coverage` check reruns all tests without optimizations though, so if there was an error we would see it there... but it's not ideal.

I'm considering that we might want to run the test suite with and without optimizations...

Reply:
> This is saying that with optimizations enabled the custom errors are not checked, right?

Yes, but it wasn't before anyway. I'd like to fix it though. Just not sure how and we might require the ABI

> I'm considering that we might want to run the test suite with and without optimizations...

Is this for always checking thoroughly? I'd like to explore an in-code fix (eg. exploring an alternative message decoding)

Reply:
I removed this block and the CI is passing, so let's solve this